Fix #39 with new log files management

bcornec opened the pull request:
- Default location for log files is now under the HOME directory of the
  user instead of /var/log
- Create that default .redfish dir if not already existing
- Adapt error messages

bcornec committed:
Fix #39 with new log files management

- Default location for log files is now under the HOME directory of the
  user instead of /var/log
- Create that default .redfish dir if not already existing
- Adapt error messages

bcornec committed:
Fix #40 by using the .redfish dir for inventory

change references to .redfish.conf into .redfish.inventory

uggla commented on redfish/config.py:
```diff
@@ -16,7 +16,22 @@
+HOME = os.getenv('HOME')
+if HOME == '':
+    print("$HOME environment variable not set, please check your system")
+    sys.exit(1)
```
ok, so we should remove that check in the redfish-client to be DRY.

uggla commented on redfish/config.py:
```diff
@@ -16,7 +16,22 @@
+        print('       Try to create directory {}'.format(os.path.dirname(REDFISH_LOGFILE)))
+        print('       using: mkdir -p {}'.format(os.path.dirname(REDFISH_LOGFILE)))
+        sys.exit(1)
+
```
This part should be in the **init** method, or a function called here and at init step.
The code placed here will be executed at the import of the module, so it is fine for the default location but if the user is setting another location for the logs using the redfish-client options, no check will be done and it will fail.

uggla commented on redfish/config.py:
```diff
@@ -16,7 +16,22 @@
+        print('       using: mkdir -p {}'.format(os.path.dirname(REDFISH_LOGFILE)))
+        sys.exit(1)
+
+REDFISH_LOGFILE = REDFISH_HOME + "/python-redfish.log"
```
Good practive is to use os.path.join() to join path.
It deals with trailing slash or various thing that could lead to double // or invalid path. Although in this case it is pretty safe.

bcornec replied:
If it's a good practice, it then wasn't applied here as I copied what existed and adapted ;-)

uggla commented on redfish/config.py:
```diff
@@ -43,14 +58,13 @@ def initialize_logger(REDFISH_LOGFILE,
         )
+    f = open(os.path.expandvars(REDFISH_LOGFILE), 'w')
+    f.close()
+
```
To my mind it is not needed. I guess you want to create the file, but that will be handled by RotatingFileHandler.
BTW, it is good practive to use the context manager "with" to open files. (I could explain you the benefits)

bcornec replied:
Without this, in case the file doesn't exist yet, it's not created by RotatingFileHandler. I experienced that thus added these 2 lines. I'm not sure I undertand why we need a with here, but let's discuss this afternoon.

uggla commented:
I have tried to explain some concerns with this PR. However that's not so easy.
So if it is unclear maybe we'll be able to handle this tomorrow with our F2F meeting.

Anyway I feel having the log by default in the home dir is good !

bcornec committed:
Improve #39

- manage variable expanion for logfiles
- improve inventory handling
- remove now useless checks for HOME
